### PR TITLE
fix to use http_proxy for git clone

### PR DESCRIPTION
**Pull request by yashsinghcodes**

### CODE REVIEW

**Comment by yashsinghcodes** [comment]
fix for #1447

File: backend/go-app/main.go [hunk 1]
```diff
@@ -397,6 +397,52 @@ func checkUsername(Username string) error {
 	return nil
 }
 
+func isNoProxy(rawURL string) bool {
```
Comment (frikky) [inline backend/go-app/main.go:400]: Give it a Git related noproxy name if that what it is used for :P

Comment (yashsinghcodes) [reply backend/go-app/main.go:400]: but it could be use for any proxy condition, no? To check if the url is part of NO_PROXY? Not specifically for git.

Comment (frikky) [reply backend/go-app/main.go:400]: Isn't that handled by the default library if we implement NO_PROXY with the default proxy HTTP config? 

https://cs.opensource.google/go/x/net/+/refs/tags/v0.28.0:http/httpproxy/proxy.go;l=51

Comment (yashsinghcodes) [reply backend/go-app/main.go:400]: The git library does not by default but that's actually smart to just force git to use a `net/http` Client which provide us to use `http.ProxyFromEnvironment`.

**Comment by yashsinghcodes** [comment]
for git proxy:
https://github.com/user-attachments/assets/0a836328-d656-4ec1-94b0-c8c8223d7c44

for no git proxy:
https://github.com/user-attachments/assets/55e972c3-d267-45f9-86f5-92beeff6873d


@frikky :)

**Pull request merged**